build.sbt

```diff
@@ -259,169 +234,320 @@ lazy val coreJVMTests = Project(id = "coreJVMTests", base = file("enumeratum-cor
       scalaVersion,
       sbtVersion,
       BuildInfoKey.action("macrosJVMClassesDir") {
-        ((macrosJVM / classDirectory) in Compile).value
+        (macrosJVM / Compile / classDirectory).value
       }
     ),
     buildInfoPackage := "enumeratum"
   )
-  .settings(commonWithPublishSettings: _*)
-  .settings(testSettings: _*)
+  .settings(commonWithPublishSettings)
+  .settings(testSettings)
   .settings(
     name               := "coreJVMTests",
     version            := Versions.Core.stable,
     crossScalaVersions := scalaVersionsAll,
-    libraryDependencies ++= Seq(
-      "org.scala-lang" % "scala-compiler" % scalaVersion.value % Test
-    ),
+    Test / sourceGenerators += CoreJVMTest.testsGenerator,
+    libraryDependencies += {
+      if (scalaBinaryVersion.value == "3") {
+        "org.scala-lang" %% "scala3-compiler" % scalaVersion.value % Test
+      } else {
+        "org.scala-lang" % "scala-compiler" % scalaVersion.value % Test
+      }
+    },
+    libraryDependencies += scalaXmlTest.value,
     publishArtifact := false,
     publishLocal    := {}
   )
   .dependsOn(coreJVM, macrosJVM)
 
+lazy val scalaXmlTest = Def.setting[ModuleID] {
+  val ver: String = {
+    if (scalaBinaryVersion.value == "2.11") "1.3.0"
+    else "2.1.0"
+  }
+
+  "org.scala-lang.modules" %% "scala-xml" % ver % Test
+}
```

Comment:
Saw this dep. being manually added to a bunch of places but not quite sure why; can you shed some light ? .

Reply:
It's re-added in `Test` scope as exclude from dependencies at `compile` scope as when present there together with `cross(CrossVersion.for3Use2_13)` (still required for some modules), it leads to have this XML modules in different version for different Scala version so ...